fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalCheckPolicy.java

```diff
@@ -0,0 +1,177 @@
+import org.apache.doris.policy.PolicyMgr;
+import org.apache.doris.policy.RowPolicy;
+import org.apache.doris.qe.ConnectContext;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Logical Check Policy
+ */
+public class LogicalCheckPolicy<CHILD_TYPE extends Plan> extends LogicalUnary<CHILD_TYPE> {
+
+    public LogicalCheckPolicy(CHILD_TYPE child) {
+        super(PlanType.LOGICAL_CHECK_POLICY, child);
+    }
+
+    public LogicalCheckPolicy(Optional<GroupExpression> groupExpression,
+            Optional<LogicalProperties> logicalProperties, CHILD_TYPE child) {
+        super(PlanType.LOGICAL_CHECK_POLICY, groupExpression, logicalProperties, child);
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitLogicalCheckPolicy(this, context);
+    }
+
+    @Override
+    public List<? extends Expression> getExpressions() {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public List<Slot> computeOutput() {
+        return child().getOutput();
+    }
+
+    @Override
+    public String toString() {
+        return Utils.toSqlString("LogicalCheckPolicy",
+            "child", child()
+        );
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        LogicalCheckPolicy that = (LogicalCheckPolicy) o;
+        return child().equals(that.child());
+    }
+
+    @Override
+    public int hashCode() {
+        return child().hashCode();
+    }
+
+    @Override
+    public Plan withGroupExpression(Optional<GroupExpression> groupExpression) {
+        return new LogicalCheckPolicy<>(groupExpression, Optional.of(getLogicalProperties()), child());
+    }
+
+    @Override
+    public Plan withLogicalProperties(Optional<LogicalProperties> logicalProperties) {
+        return new LogicalCheckPolicy<>(Optional.empty(), logicalProperties, child());
+    }
+
+    @Override
+    public Plan withChildren(List<Plan> children) {
+        Preconditions.checkArgument(children.size() == 1);
+        return new LogicalCheckPolicy<>(children.get(0));
+    }
+
+    /**
+     * get wherePredicate of policy for logicalRelation.
+     *
+     * @param logicalRelation include tableName and dbName
+     * @param connectContext include information about user and policy
+     */
+    public Optional<Expression> getFilter(LogicalRelation logicalRelation, ConnectContext connectContext) {
+        String dbName = !logicalRelation.getQualifier().isEmpty() ? logicalRelation.getQualifier().get(0) : null;
+        Database db = connectContext.getEnv().getInternalCatalog().getDb(dbName)
+                .orElseThrow(() -> new RuntimeException("Database [" + dbName + "] does not exist."));
```

Comment:
The `LogicalTVFRelation` is `LogicalRelation` but should not throw exception.
I think we should provide some interfaces, like `CatalogRelation`, `InternalCatalogRelation`, `ExternalCatalogRelation`, then we check for `CatalogRelation`